pkg/apps/blackduck/components/configmap/database/v1/configmap.go

```diff
@@ -1,61 +1,65 @@
 package v1
 
 import (
+	"strconv"
+
 	horizonapi "github.com/blackducksoftware/horizon/pkg/api"
 	"github.com/blackducksoftware/horizon/pkg/components"
 	blackduckapi "github.com/blackducksoftware/synopsys-operator/pkg/api/blackduck/v1"
-	"github.com/blackducksoftware/synopsys-operator/pkg/apps/blackduck/types"
+	"github.com/blackducksoftware/synopsys-operator/pkg/apps/blackduck"
 	"github.com/blackducksoftware/synopsys-operator/pkg/apps/store"
+	"github.com/blackducksoftware/synopsys-operator/pkg/apps/types"
 	apputils "github.com/blackducksoftware/synopsys-operator/pkg/apps/utils"
 	"github.com/blackducksoftware/synopsys-operator/pkg/protoform"
 	"github.com/blackducksoftware/synopsys-operator/pkg/util"
 	"k8s.io/client-go/kubernetes"
-	"strconv"
 )
 
+// BdConfigmap holds the Black Duck config map configuration
 type BdConfigmap struct {
 	config     *protoform.Config
 	kubeClient *kubernetes.Clientset
-	blackduck  *blackduckapi.Blackduck
+	blackDuck  *blackduckapi.Blackduck
 }
 
-func NewBdConfigmap(config *protoform.Config, kubeClient *kubernetes.Clientset, blackduck *blackduckapi.Blackduck) types.ConfigMapInterface {
-	return &BdConfigmap{config: config, kubeClient: kubeClient, blackduck: blackduck}
+// NewBdConfigmap returns the Black Duck config map configuration
+func NewBdConfigmap(config *protoform.Config, kubeClient *kubernetes.Clientset, blackDuck interface{}) types.ConfigMapInterface {
+	return &BdConfigmap{config: config, kubeClient: kubeClient, blackDuck: blackDuck.(*blackduckapi.Blackduck)}
```

Comment:
This will cause a panic if we pass an object that isn't of type Blackduck. We should probably check the type and return an error if it's not what we expect. Same thing for the other constructors.

Reply:
@jdartigalongue The below method will already take care of it.. Only if it is of type, then only it will apply to its resource. Is it fine or still you want to check the type?

func getCR(cr interface{}) (*customResource, error) {
	switch reflect.TypeOf(cr) {
	case reflect.TypeOf(&alertapi.Alert{}):
		obj := cr.(*alertapi.Alert)
		return &customResource{namespace: obj.Spec.Namespace, size: obj.Spec.Size, imageRegistries: obj.Spec.ImageRegistries, registryConfiguration: &obj.Spec.RegistryConfiguration}, nil
	case reflect.TypeOf(&blackduckapi.Blackduck{}):
		obj := cr.(*blackduckapi.Blackduck)
		return &customResource{namespace: obj.Spec.Namespace, size: obj.Spec.Size, imageRegistries: obj.Spec.ImageRegistries, registryConfiguration: &obj.Spec.RegistryConfiguration}, nil
	case reflect.TypeOf(&opssightapi.OpsSight{}):
		obj := cr.(*opssightapi.OpsSight)
		return &customResource{namespace: obj.Spec.Namespace, size: obj.Spec.Size, imageRegistries: obj.Spec.ImageRegistries, registryConfiguration: &obj.Spec.RegistryConfiguration}, nil
	}
	return nil, fmt.Errorf("unable to handle the custom resource")
}